.jenkins/Jenkinsfile

```diff
@@ -0,0 +1,77 @@
+// oetools-image:tag  Docker image from OE Jenkins Registry
+OETOOLS_IMAGE = "oejenkinscidockerregistry.azurecr.io/oetools-azure:1.7"
+
+// Tests running on hardware with custom path to libdcap_quoteprov.so
+def coffeelakeTest(String compiler, String suite) {
+    stage("Coffeelake $compiler SGX1-FLC $suite") {
+        node('hardware') {
+            checkout scm
+            
+            // Generate libdcap_quoteprov.so in the $WORKSPACE/src/Linux folder
+            def buildImage = docker.build("az-dcap-builder", '.jenkins')
+            buildImage.inside {
+                dir('src/Linux') {
+                    sh './configure'
+                    sh 'make'
+                }
+            }
+            
+            // Run hardware tests using custom LD_PRELOAD
+            dir('openenclave') {
+                git url: 'https://github.com/Microsoft/openenclave.git'
+                timeout(15) {
+                    sh "LD_PRELOAD=$WORKSPACE/src/Linux/libdcap_quoteprov.so ./scripts/test-build-config -p  SGX1FLC -b $suite -d --compiler=$compiler"
+                }
+            }
+        }
+    }
+}
+
+// Test using oetools-test Docker image with /dev/sgx mounted inside container
+def nonSimulationTest() {
+    stage('Non-Simulation Container SGX1-FLC RelWithDebInfo') {
+        node('hardware') {
+            checkout scm
+            
+            // build az-dcap-client deb package
+            def buildImage = docker.build("az-dcap-builder", '.jenkins')
+            buildImage.inside {
+                dir('src/Linux') {
+                    sh 'dpkg-buildpackage -us -uc' 
+                }
+            }
+
+            // Clone openenclave repo
+            dir('openenclave') {
+                git url: 'https://github.com/Microsoft/openenclave.git'
+            }
+            /*
+            Use oetools-azure image from OE Jenkins Registry
+            Remove the installed az-dcap-client from the container
+            Install az-dcap-client in the container from the deb package we previously built
+            */
+            docker.image(OETOOLS_IMAGE).inside('-u root --privileged -v /dev/sgx:/dev/sgx') {
+                dir('src') {
+                    sh 'apt remove -y az-dcap-client'
+                    sh 'dpkg -i *.deb'
+                }
+                dir('openenclave') {
+                    timeout(15) {
+                        sh './scripts/test-build-config -p SGX1FLC -b RelWithDebInfo -d'
+                        // We need to remove the build folder because is created as root
+                        sh 'rm -rf ./build'
```

Comment:
Shouldn't the build folder automatically go away when the container is terminated?

Reply:
Jenkins is automatically mounting the Workspace folder inside the container.

Here we are using the oetools-azure image built from openenclave project.
That image though , has the az-dcap-client already installed.   So I am doing `apt remove -y az-dcap-client`  and then Install the current version from the deb that we build.
In order to remove and install the new lib , the Non-Simulation test needs to run the docker container as root

Running the test-build-config  script after that will throw errors if I don't delete the build folder created as root
https://github.com/Microsoft/openenclave/blob/master/scripts/test-build-config#L123-L127